haystack/generic_views.py

```diff
@@ -88,9 +88,10 @@ class FacetedSearchMixin(SearchMixin):
     faceting.
     """
     form_class = FacetedSearchForm
+    facet_fields = None
```

Comment:
Thanks for the patch. I used this update within my project as a work around for the bug.

I also liked the idea of adding facet_fields attribute to view. For my project I needed to pass additional args for sorting and limits.

How about allowing this to be an tuple of dicts, then unpack each dict for the args in the facet method?

```python
facet_fields = ({'field': 'author', 'sort': 'index', 'limit': -1},)
```

```python
def get_queryset(self):
    qs = super(FacetedSearchMixin, self).get_queryset()
    for field in self.facet_fields:
        qs = qs.facet(**field)
    return qs
```